### PR TITLE
User can cancel registration

### LINKED ISSUE #1636: Can't cancel registration
Steps:

- Log in as a non-registered-to-bid user
- Open a live auction
- Tap "register to bid"
- Decide _not_ to register
- There's no "cancel" button 

The mobile web view controller used for registration expects to be presented as a push, not modally. We need to wrap it in a serif nav controller.

Note: I thought I saw this issue already reported, but couldn't find it. Link+close if you do.

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes #1636.

### CODE REVIEW

**Comment by orta** [comment]
This'll have the downside of looking like:

screen shot 2016-06-07 at 4 19 17 pm

( Wherein it has a big white bar across the top )

**Comment by orta** [comment]
( Was having the same issue with #1533 )

**Comment by orta** [comment]
I feel like we need a `HostedWebViewModalNavigationController`  thing

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            1 Warning
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |238.7ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Artsy/View_Controllers/Auction/AuctionViewController.swift#L181) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
179.9ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
161.6ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
71.9ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
66.0ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
63.1ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, sideMargin: String)
59.6ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
57.8ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-reg-cancel/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)




<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
Okay, I think I got this:

screen shot 2016-06-07 at 5 39 06 pm
screen shot 2016-06-07 at 5 26 00 pm

Looking forward to feedback 👍

File: Artsy/View_Controllers/SerifModalWebNavigationController.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    override func shouldAutorotate() -> Bool {
+        return traitDependentAutorotateSupport
+    }
+
+    func dismiss() {
+        presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
+    }
+
+    func navigationController(navigationController: UINavigationController, willShowViewController viewController: UIViewController, animated: Bool) {
+        viewController.automaticallyAdjustsScrollViewInsets = false
```
Comment (orta) [inline Artsy/View_Controllers/SerifModalWebNavigationController.swift]: ah, interesting

**Comment by orta** [comment]
Yep!

**Pull request merged**